controller/users.go

```diff
@@ -102,6 +102,31 @@ func copyExistingKeycloakUserProfileInfo(existingProfile *login.KeycloakUserProf
 	return keycloakUserProfile
 }
 
+func (c *UsersController) populateExistingKeycloakProfileData(ctx *app.UpdateUsersContext, keycloakUserProfile *login.KeycloakUserProfile) error {
+
+	// ensure we dont call KC repeatedly
+	if keycloakUserProfile.Username != nil {
```

Comment:
I'm not following this logic... This method will be called multiple times anyway. And all calls are inside a DB transactions...